sequencer.c

```diff
@@ -2524,7 +2524,8 @@ static int create_seq_dir(void)
 {
 	if (file_exists(git_path_seq_dir())) {
 		error(_("a cherry-pick or revert is already in progress"));
-		advise(_("try \"git cherry-pick (--continue | --quit | --abort)\""));
+		advise(_("try \"git cherry-pick (--continue | --skip | --quit | --abort)\""));
+		advise(_("or  \"git revert (--continue | --skip | --quit | --abort)\""));
```

Comment:
I think this change deserves a mention in the commit message, if not its own commit.  It is not directly related to the changes in this commit.  Maybe something like "While there, suggest using `git cherry-pick --<sequencer-command>` or `git revert --<sequencer-command>` where we don't know whether a revert or a cherry-pick is in progress." 

I think I'd prefer adding the `git revert` advice (without `--skip`) as the very first patch in the series though. (Not a strong preference though)

Reply:
> I think I'd prefer adding the `git revert` advice (without `--skip`) as the very first patch in the series though.

I agree. This would make for a nicer story to follow (I like to think of patch series as stories that developers tell other developers).

Reply:
@tgummerer @newren @dscho Yes, even better, why not introduce a preparatory step which will introduce this advice and `sequencer_skip` function? This will make it a three patch story.

1. Introduce `sequencer_skip` and advice
2. Use `sequencer_skip` and introduce new tests
3. Change advice messages

That will also make individual patches small and easier to review

Reply:
The additional advice and the `sequencer_skip` don't seem related to me. Introducing sequencer_skip separately in an additional commit is an option though.

Reply:
Your #1 combines two things that are unrelated in my mind; you should split those apart.  I'm also slightly worried about introducing sequencer_skip in a separate commit from using it; that can cause compilation errors (unused function; DEVELOPER=1 implies -Werror).  It may be better to include it in the same commit that makes use of the new function.

I think the documentation for --skip should be included in the same commit that adds the option.  Your plan to put the changes to advice strings (from mentioning 'git reset' to instead use --skip with cherry-pick or revert) seems fine.

Reply:
Now that I think about it deeply, it makes sense to not introduce advice about revert in [# 1](https://github.com/r1walz/git/pull/1#discussion_r288657829), but setting it up in # 2 and changing it in # 3. This flow looks better and makes more sense to me.

Reply:
@newren If we add the documentation in the first commit and not change the advice, the user might find it more confusing that documentation mentions `--skip` flag but git advises using `git reset` leaving them even more confused? The commits are tightly coupled as they are.

Reply:
The choice of where to split commits is irrelevant to end users, who will be using your whole series or none of it.  Splitting commits is about explaining to other developers what changes are made and why.

In particular, you are adding a new way to access the 'skip' functionality in one commit, and then in the second commit you are suggesting people use it instead of the old and confusing mechanism.

Said another way, I'm reiterating what Thomas said about a week ago in this PR, when he said you should have "A commit introducing the new feature, *including* documentation and the new tests" (emphasis added).

Reply:
My preferred 3-chapter story would have been:

1. introduce the advice mentioning `git revert` (without `--skip`)
2. support `--skip` in `git cherry-pick` *and* document it
3. change the advice to recommend `--skip` instead of `git reset`